framework/src/loops/ComputeMortarFunctor.C

```diff
@@ -181,7 +185,10 @@ ComputeMortarFunctor::operator()()
     // reinit the variables/residuals/jacobians on the lower dimensional element corresponding to
     // the slave face. This must be done last after the dof indices have been prepared for the
     // slave (element) and master (neighbor)
-    _subproblem.reinitLowerDElemRef(slave_face_elem, &custom_xi1_pts);
+    if (_use_dual)
+      _subproblem.reinitLowerDElemDualRef(slave_face_elem, &custom_xi1_pts);
+    else
+      _subproblem.reinitLowerDElemRef(slave_face_elem, &custom_xi1_pts);
```

Comment:
It would be nice to just have a single `reinitLowerDElemRef` call that would initialize both "standard" and "dual" data structures. I feel like we could do that...

Reply:
> It would be nice to just have a single `reinitLowerDElemRef` call that would initialize both "standard" and "dual" data structures. I feel like we could do that...

Okay, so you mean in `Assembly::reinitLowerDElemRef` do something like: `if (_use_dual) fe_lower->get_dualphi() ... else fe_lower->get_phi()', right?

Reply:
> It would be nice to just have a single `reinitLowerDElemRef` call that would initialize both "standard" and "dual" data structures. I feel like we could do that...

I am not sure if we will need to initialize both of them as we probably will (mostly) only be using one of them.. 

But I do not mind initializing both as that would be a easier implementation